### PR TITLE
New Package: btop-1.2.0

### LINKED ISSUE #33081: Package Request: btop++ (bpytop replacement *in future when it reaches the same functionality*)
An updated rewrite of bpytop, in c++.

https://github.com/aristocratos/btop

### PR DESCRIPTION
**Pull request by RunningDroid**
I'll let someone else decide whether or not adding this to the repos means we should rm bpytop.

closes #33081

#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me

### CODE REVIEW

**Comment by dmarto** [comment]
_just a passing comment from a bpytop user_
I would prefer if `bpytop` is left for the foreseeable future, as `btop++` is still under heavy development and there may be feature parity, but there isn't "bug parity" if you catch my point.

**Comment by RunningDroid** [comment]
My latest force-push makes it build for non-x86 archs, but it isn't pulling in CFLAGS or LDFLAGS for some reason, I'll work on it some more tomorrow.

**Comment by RunningDroid** [comment]
I updated this to 1.0.13, and the latest force-push fixes CXXFLAGS and LDFLAGS not getting pulled in.

File: srcpkgs/btop/template [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+build_style=gnu-makefile
+make_build_args="ARCH=\${XBPS_TARGET_MACHINE} ADDFLAGS=\${CXXFLAGS}\${LDFLAGS}"
+make_use_env=true
+short_desc="Monitor of resources"
+maintainer="RunningDroid <runningdroid@zoho.com>"
+license="Apache-2.0"
+homepage="https://github.com/aristocratos/btop"
+distfiles="https://github.com/aristocratos/btop/archive/refs/tags/v${version}.tar.gz"
+checksum=e92da56853c105d3fe62a5b175c1bd3b9a9ce69b9b9a274f0c5f2ad21363c7e2
+make_check=no # no tests
```
Comment (paper42) [inline srcpkgs/btop/template:14]: Why is this required? When a package doesn't have tests, they are automatically skipped.

File: srcpkgs/btop/template [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Template file for 'btop'
+pkgname=btop
+version=1.0.24
```
Comment (paper42) [inline srcpkgs/btop/template:3]: 1.1.3 is out

File: srcpkgs/btop/template [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+revision=1
+build_style=gnu-makefile
+make_build_args="ARCH=\${XBPS_TARGET_MACHINE} ADDFLAGS=\${CXXFLAGS}\${LDFLAGS}"
+make_use_env=true
+short_desc="Monitor of resources"
+maintainer="RunningDroid <runningdroid@zoho.com>"
+license="Apache-2.0"
+homepage="https://github.com/aristocratos/btop"
+distfiles="https://github.com/aristocratos/btop/archive/refs/tags/v${version}.tar.gz"
```
Comment (paper42) [inline srcpkgs/btop/template:12]:
```suggestion
homepage="https://github.com/aristocratos/btop"
changelog="https://raw.githubusercontent.com/aristocratos/btop/main/CHANGELOG.md"
distfiles="https://github.com/aristocratos/btop/archive/refs/tags/v${version}.tar.gz"
```

File: srcpkgs/btop/template (hunk 1 above)
Comment (RunningDroid) [reply srcpkgs/btop/template:14]: I had just added it to silence the warning, but I'll remove it.

File: srcpkgs/btop/template (hunk 2 above)
Comment (RunningDroid) [reply srcpkgs/btop/template]: I'd been procrastinating updating this because 1.1.0 broke cross and I haven't had time to work out how to fix it, I've pushed an updated template so it's clear that it's (hopefully temporarily) broken.

**Comment by RunningDroid** [comment]
I tried to see if any other distros had patches but it looks like they just disabled building on non-x86 archs, so I've done the same for now.

**Comment by prateekmedia** [comment]
@paper42 What's blocking this PR?

**Comment by paper42** [comment]
It compiles for all architectures, so I removed the arch restriction.
